test/src/main/java/org/axonframework/test/saga/RecordingLoggingErrorHandler.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.test.saga;
+
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.EventMessageHandler;
+import org.axonframework.eventhandling.LoggingErrorHandler;
+
+import java.util.Optional;
+
+public class RecordingLoggingErrorHandler extends LoggingErrorHandler {
```

Comment:
I'm in doubt whether this should be a `LoggingErrorHandler`.
Wouldn't it be nicer if this is a wrapper around another `ListenerInvocationErrorHandler`? That we, registering another `ListenerInvocationErrorHandler` is supported, as well as recording the failure.